src/codeflare_sdk/cluster/cluster.py

```diff
@@ -137,6 +144,9 @@ def down(self):
         associated with the cluster.
         """
         namespace = self.config.namespace
+        if self.app_wrapper_yaml is None:
+            self.app_wrapper_yaml = self.create_app_wrapper()
```

Comment:
Is there another way to get `self.app_wrapper_name` here other than generating the appwrapper yaml. Seems unnecessary to generate the appwrapper yaml just to delete the resource.